manifests/init.pp

```diff
@@ -3,75 +3,100 @@
 #
 # Contributor(s) : Remi Ferrand <remi.ferrand_at_cc(dot)in2p3(dot)fr>
 #
-
-# == Define: etc_services
+# @summary Manage a /etc/services entry uniquely identified by its name and protocol.
+#
+# @param service_name [String]
+#   The name of the service in /etc/services. This is a namevar...
+#   Note that it should comply with the syntax laid out in 
+#   [RFC 6335 Section 5.1](https://tools.ietf.org/html/rfc6335#section-5.1)
+#
+# @param enforce_syntax [Boolean]
+#   When set to true the syntax rules from RFC 6335 are enforced.
+#
+# @param protocols [Hash[Enum['tcp','udp'],Integer]]
+#   A hash mapping one or more protocols to their associated ports. This is
+#   mandatory.
 #
-# Manage a /etc/services entry uniquely identified by its name and protocol.
+# @param comment [String]
+#   An optional comment to be appended to the end of each port/protocol
+#   specific line in /etc/services.
 #
-# === Parameters
+# @param aliases [Array[String]]
+#   An optional array of aliases which will be included for each port/protocol
+#   combination in /etc/services.
 #
-# [*port*]
-#   /etc/services entry port. Required.
-# [*comment]
-#   /etc/services entry comment. Defaults to ''.
-# [*aliases*]
-#   /etc/services entry aliases specified as an array. Defaults to [].
-# [*ensure*]
-#   Should /etc/services entry be present or absent. Defaults to present.
+# @param ensure [Enum['absent','present']]
+#   Should the corresponding /etc/services entry/entries be present or absent?
 #
 define etc_services (
-  $port,
-  $comment = '',
-  $aliases = [],
-  $ensure = 'present'
+  String $service_name                       = $name,
+  Boolean $enforce_syntax                    = true,
+  Enum['absent','present'] $ensure           = 'present',
+  String $comment                            = '',
+  Array[String] $aliases                     = [],
+  Hash[Enum['tcp','udp'],Integer] $protocols = undef,
 )
 {
-  validate_re($name, '^[-\w]+/(tcp|udp)$')
-  validate_re($ensure, '^(absent|present)$')
-  validate_re($port, '^\d+$')
-  validate_array($aliases)
-  validate_string($comment)
+  # Validate the name per RFC 6335 section 5.1
+  # 1-15 characters
+  # Begins and ends with [A-Za-z0-9]
+  # Includes only [A-Za-z0-9-]
+  # No consecutive '-'
+  if $enforce_syntax {
+    unless($service_name =~
+      /^(?=.{1,15}$)(?=[A-Za-z0-9])(?=[A-Za-z0-9-]*[A-Za-z0-9]$)(?!.*([-])\1)[A-Za-z0-9-]+$/) {
```

Comment:
Suggestion: we could turn this regex into its own data type

```puppet
unless $service_name =~ Etc_services::Service_name
```

Reply:
Sure, I'll have to refresh my memory on the changes as it's been a bit since I looked at the code. Definitely a good call on the type; this was one of (possibly *the* first) early modules I modified and I honestly didn't know about types at the time.

Reply:
if it's too far away I'll give it a shot, it's an easy change

Reply:
Oh, I meant the "how to migrate" paragraph as I really had no memory of how it worked originally. Turns out the change was pretty simple.

Reply:
so, in the end do you need help on this ?

Reply:
No, it's already done. Please check the latest commits instead of this portion.